src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs

```diff
@@ -104,6 +106,19 @@ public bool IsBrokerInstalledAndInvokable()
             }
         }
 
+        private void ValidateRedirectUri(Uri redirectUri)
+        {
+            string bundle = NSBundle.MainBundle.BundleIdentifier;
+            string expectedRedirectUri = $"msauth.{bundle}://auth";
+            if (!string.Equals(expectedRedirectUri, redirectUri.AbsoluteUri, StringComparison.OrdinalIgnoreCase))
```

Comment:
@antrix1989 - could you please confirm the logic here? I want to throw if the user sets a redirect uri that does not follow this rule, although I've seen that in practice this rule is not set in stone, i.e. ADAL style redirect URIs continue to work. Note that we do not use SSO extensions yet.